src/Xrefcheck/Scanners/Markdown.hs

```diff
@@ -155,40 +178,49 @@ cataNodeWithParentNodeInfo f node = cataNode f' node
       map (ssParentNodeType .= Just ty >>) childScanners
 
 -- | Find ignore annotations (ignore paragraph and ignore link)
--- and remove nodes that should be ignored.
-processAnnotations :: FilePath -> C.Node -> Writer [ScanError] C.Node
+-- and remove nodes that should be ignored;
+-- find copy/paste check annotations (ignore for paragraph and for link)
+-- and label nodes with a boolean meaning whether they should be
+-- copy/paste checked.
+processAnnotations :: FilePath -> C.Node -> Writer [ScanError] NodeCPC
```

Comment:
All this code seems to not scale pretty well (in LOC) with new kinds of annotations added. Also, as we have been discussing, it is difficult to reason about how they overlap or affect each other.

But it is such a big topic that we could leave it for another issue, in which we try to figure out some clear abstraction over Xrefcheck annotations.